packages/grid/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -0,0 +1,256 @@
+      });
+
+      renderedRowsIntervalCache.current = params;
+
+      const fetchRowsParams: GridFetchRowsParams = {
+        firstRowToRender: firstRowIndex,
+        lastRowToRender: lastRowIndex,
+        sortModel,
+        filterModel,
+      };
+
+      apiRef.current.publishEvent('fetchRows', fetchRowsParams);
+    },
+    [
+      apiRef,
+      props.rowsLoadingMode,
+      rowIds,
+      sortModel,
+      filterModel,
+      visibleRows,
+      props.experimentalFeatures,
+    ],
+  );
+
+  const handleGridSortModelChange = React.useCallback<GridEventListener<'sortModelChange'>>(
+    (newSortModel) => {
+      const dimensions = apiRef.current.getRootDimensions();
+      const { lazyLoading } = (props.experimentalFeatures ?? {}) as GridExperimentalProFeatures;
+
+      if (
+        isLazyLoadingDisabled({
+          lazyLoadingFeatureFlag: lazyLoading,
+          rowsLoadingMode: props.rowsLoadingMode,
+          gridDimentions: dimensions,
+        })
+      ) {
+        return;
+      }
+
+      const currentRenderContext = apiRef.current.unstable_getRenderContext();
+      const [firstRowToRender, lastRowToRender] = getRenderableIndexes({
+        firstIndex: currentRenderContext.firstRowIndex,
+        lastIndex: currentRenderContext.lastRowIndex,
+        minFirstIndex: 0,
+        maxLastIndex: visibleRows.rows.length,
+        buffer: props.rowBuffer,
+      });
+      const fetchRowsParams: GridFetchRowsParams = {
+        firstRowToRender,
+        lastRowToRender,
+        sortModel: newSortModel,
+        filterModel,
+      };
+
+      apiRef.current.publishEvent('fetchRows', fetchRowsParams);
+    },
+    [
+      apiRef,
+      props.rowsLoadingMode,
+      props.rowBuffer,
+      visibleRows.rows,
+      filterModel,
+      props.experimentalFeatures,
+    ],
+  );
+
+  const handleGridFilterModelChange = React.useCallback<GridEventListener<'filterModelChange'>>(
+    (newFilterModel) => {
+      const dimensions = apiRef.current.getRootDimensions();
+      const { lazyLoading } = (props.experimentalFeatures ?? {}) as GridExperimentalProFeatures;
+
+      if (
+        isLazyLoadingDisabled({
+          lazyLoadingFeatureFlag: lazyLoading,
+          rowsLoadingMode: props.rowsLoadingMode,
+          gridDimentions: dimensions,
+        })
+      ) {
+        return;
+      }
+
+      const currentRenderContext = apiRef.current.unstable_getRenderContext();
+      const [firstRowToRender, lastRowToRender] = getRenderableIndexes({
+        firstIndex: currentRenderContext.firstRowIndex,
+        lastIndex: currentRenderContext.lastRowIndex,
+        minFirstIndex: 0,
+        maxLastIndex: visibleRows.rows.length,
+        buffer: props.rowBuffer,
+      });
+      const fetchRowsParams: GridFetchRowsParams = {
+        firstRowToRender,
+        lastRowToRender,
+        sortModel,
+        filterModel: newFilterModel,
+      };
+
+      apiRef.current.publishEvent('fetchRows', fetchRowsParams);
```

Comment:
Could part of `handleRenderedRowsIntervalChange` be reused here? Same for `handleGridSortModelChange`. This code seems duplicated.

Reply:
`handleRenderedRowsIntervalChange ` does more than it's needed for the filtering and sorting cases.
You are right, the code is very similar, the difference is that in the filtering case the filter model that is being passed is from the function parameters and so the sorting model is from the state and vice versa for the sorting case.
I can make it into a single event handler but having them separated makes it easier to read and reason.